sales/src/execute.rs

```diff
@@ -0,0 +1,209 @@
+use std::{cell::RefCell, rc::Rc};
+
+use burnt_glue::response::Response;
+use cosmwasm_std::{
+    BankMsg, Coin, CosmosMsg, CustomMsg, Deps, DepsMut, Env, MessageInfo, Timestamp, Uint64,
+};
+use cw721_base::{state::TokenInfo, MintMsg};
+use cw_storage_plus::Item;
+use ownable::Ownable;
+use sellable::Sellable;
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::{errors::ContractError, state::PrimarySale, Sales};
+
+impl<'a, T, C, E, Q> Sales<'a, T, C, E, Q>
+where
+    T: Serialize + DeserializeOwned + Clone,
+    Q: CustomMsg,
+    E: CustomMsg,
+    C: CustomMsg,
+{
+    pub fn new(
+        sellable_module: Rc<RefCell<Sellable<'a, T, C, E, Q>>>,
+        primary_sales: Item<'a, Vec<PrimarySale>>,
+    ) -> Self {
+        Self {
+            sellable: sellable_module,
+            primary_sales,
+        }
+    }
+
+    pub fn add_primary_sales(
```

Comment:
This could simply accept a `ExecuteMsg::PrimarySale` object

Reply:
To do this you will have to pull out the contents of ExecuteMsg::PrimarySale into a new struct e.g. `pub struct CreatePrimarySale` and then refactor the enum to wrap it, e.g. `PrimarySale(CreatePrimarySale)`

Reply:
Furthermore, we will want basic validation on the `CreatePrimarySale` object. We will want to validate that its start date is >= the current block time.

Reply:
Nice, this is cleaner